pkg/controllers/nodeclaim/lifecycle/controller.go

```diff
@@ -174,6 +174,19 @@ func (c *Controller) finalize(ctx context.Context, nodeClaim *v1.NodeClaim) (rec
 		return reconcile.Result{}, fmt.Errorf("adding nodeclaim terminationGracePeriod annotation, %w", err)
 	}
 
+	// set a disruption reason if the nodeclaim was deleted manually
+	if !nodeClaim.StatusConditions().IsTrue(v1.ConditionTypeDisruptionReason) {
```

Comment:
So I think we have a race here currently in the code with disruption and this controller fighting to add/remove the condition. We probably want to do the following:
1. When draining, assume the non-existence of the status condition indicates that it's forceful termination. 
2. In the disruption controller, check if something is terminating (has deletion timestamp set) before clearing the status condition, and only clear it if it's a condition that we own. The IsConflict() errors will prevent any races here if a node is terminated the same time it's removed from the disruption queue internally.

Ideally, we'd be able to extend this so that another forceful termination controller could choose to add its own status condition here, but since the status patch and nodeclaim delete calls can't be batched together, there's no way to guarantee the status condition for the forceful method will exist. I think doing "Forceful Termination" with correlating to other metrics are good enough here.